crates/typst/src/text/font/color.rs

```diff
@@ -6,51 +6,138 @@ use ttf_parser::{GlyphId, RgbaColor};
 use usvg::tiny_skia_path;
 use xmlwriter::XmlWriter;
 
-use crate::layout::{Abs, Axes, Frame, FrameItem, Point, Size};
+use crate::layout::{Abs, Frame, FrameItem, Point, Size};
 use crate::syntax::Span;
 use crate::text::{Font, Glyph};
-use crate::visualize::Image;
+use crate::visualize::{FixedStroke, Geometry, Image};
 
-/// Tells if a glyph is a color glyph or not in a given font.
-pub fn is_color_glyph(font: &Font, g: &Glyph) -> bool {
+/// Whether this glyph can be rendered via simple outlining instead of via
+/// `glyph_frame`.
+pub fn is_outlinable(font: &Font, glyph: &Glyph) -> bool {
     let ttf = font.ttf();
-    let glyph_id = GlyphId(g.id);
-    ttf.glyph_raster_image(glyph_id, 160).is_some()
-        || ttf.glyph_svg_image(glyph_id).is_some()
-        || ttf.is_color_glyph(glyph_id)
+    let glyph_id = GlyphId(glyph.id);
+    (ttf.tables().glyf.is_some() || ttf.tables().cff.is_some())
+        && !ttf
+            .glyph_raster_image(glyph_id, u16::MAX)
+            .is_some_and(|img| img.format == ttf_parser::RasterImageFormat::PNG)
+        && !ttf.is_color_glyph(glyph_id)
```

Comment:
`is_color_glyph` returns true if the glyph can be drawn with the `COLOR` table.  This is an outline format.